lib/airbrake-ruby/sync_sender.rb

```diff
@@ -33,18 +21,15 @@ def initialize(config)
     # @param [Airbrake::Notice] endpoint
     # @return [Hash{String=>String}] the parsed HTTP response
     def send(notice, endpoint = @config.endpoint)
-      response = nil
       req = build_post_request(endpoint, notice)
       https = build_https(endpoint)
 
-      begin
-        response = https.request(req)
-      rescue *HTTP_ERRORS => ex
-        @config.logger.error("#{LOG_LABEL} HTTP error: #{ex}")
-        return
-      end
-
+      response = https.request(req)
       Response.parse(response, @config.logger)
+
+    rescue => ex
```

Comment:
> @stevecrozz 
> I think its true that almost all of the exceptions will happen in https.request, but if any other RuntimeErrors occur, then a problem cascades upwards for users of the async sender.

True. However, how will we be able to fix it if we always rescue it? We're discussing this PR because something was not caught, so you had a chance to let us know that something is wrong.

If we rescue more than `http.request`, then in case of an error coming from elsewhere inside this method we won't be able to debug it. The execution will probably never reach `http.request` anyway and it presumably never fix itself. 

I am struggling to understand why we want to have a broken but alive worker. It has no use. `http.request`'s success depends on network, other nodes, weather, etc., and we can't predict everything, so it's wise to `rescue => ex` here, but the rest of that method is our code and our bugs.

Hope this makes sense. That said, perhaps I am misunderstanding something?